File: Solutions/007-Reverse-Integer/Solution 2.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        (sign, x)  = (1, x) if x >= 0 else (-1, abs(x))
        ret = 0
        while x != 0 :
            # In python, the negative int's % operator is different from C/C++
            ret = ret * 10 + x % 10
            x = x // 10
        ret *= sign
        return 0 if ret > 2147483647 or ret < -2147483648 else ret

File: Solutions/371  Sum of Two Integers/Solution2.cpp
class Solution {
public:
    int getSum(int a, int b) {
        int ret=0;
        int carry = 0;
        int ai,bi, s;
        for(int i = 0; i < 32; i++){
            ai = (a>>i) & 0x1;
            bi = (b>>i) & 0x1;
            s = ai ^ bi ^ carry;
            ret |= (s<<i);
            
            carry = (ai&bi) || ( (ai^bi) & carry);
        }
        
        return ret;
    }
};
